File: 3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        int s=nums.size();
        int mini=INT_MAX,maxi=INT_MIN;
        vector<int> sol;
        for(int x:nums){
            mini=min(mini,x);
            maxi=max(maxi,x);
        }
        for(int j=mini+1;j<maxi;j++){
            if (find(nums.begin(), nums.end(), j) == nums.end()) {
                sol.push_back(j);
            }
        }
        if(sol.empty()) return{};
        return sol;
    }
};

File: 3731-find-missing-elements/3731-find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        int s=nums.length;
        int mini=Integer.MAX_VALUE ;
        int maxi=Integer.MIN_VALUE ;
        List<Integer> sol = new ArrayList<>();
        for(int x:nums){
            mini=Math.min(mini,x);
            
            maxi=Math.max(maxi,x);
        }
        if(maxi - mini + 1 == s) return new ArrayList<>();
        int[] found = new int [maxi+1];
        for(int x : nums){
            found[x]++;
        }
        for(int j=mini+1;j<maxi;j++){
            if (found[j]==0) {
                sol.add(j);
            }
        }
        return sol;
    }
}